addons/website_sale/models/sale_order.py

```diff
@@ -243,6 +243,20 @@ def _update_address(self, partner_id, fnames=None):
             # Recompute taxes on fpos change
             self._recompute_taxes()
 
+        # If the user has explicitly selected a valid pricelist, we don't want to change it
+        if selected_pricelist_id := request.session.get('website_sale_selected_pl_id'):
+            selected_pricelist = (
+                self.env['product.pricelist'].browse(selected_pricelist_id).exists()
+            )
+            if (
+                selected_pricelist
+                and selected_pricelist._is_available_on_website(self.website_id)
+                and selected_pricelist._is_available_in_country(
+                    self.partner_id.country_id.code
+                )
+            ):
+                self.pricelist_id = selected_pricelist
```

Comment:
```suggestion
                self.pricelist_id = selected_pricelist
           else:
               request.session.pop('website_sale_selected_pl_id', None)
```

Let's pop it if the value is not right anymore ;)

Reply:
Sure